packages/core/src/events/dispatcher.ts

```diff
@@ -2,40 +2,44 @@ import { EventEmitter } from 'events';
 import { Events } from './types';
 
 export class EventDispatcher {
-  private readonly events: EventEmitter;
+  private readonly eventEmitter: EventEmitter;
 
-  constructor() {
-    this.events = new EventEmitter();
-    this.events.setMaxListeners(1000);
+  public constructor(eventEmitter: EventEmitter, maxListeners: number = 1000) {
+    this.eventEmitter = eventEmitter;
+    this.eventEmitter.setMaxListeners(maxListeners);
   }
```

Comment:
tady bych `EventEmitter` nepředával zvenku, je to úplně interní "helper" toho event dispatcheru, podle mě ho není potřeba injektovat

Reply:
done

Reply:
nitpick: `maxListeners` není potřeba předávat zvenku, je to vpodstatě interní setting